h/api/uri.py

```diff
@@ -64,6 +64,7 @@
 """
 import re
 
+from h._compat import text_type
```

Comment:
As touched on above, this shouldn't really be importing from `h._compat`. We should duplicate the relevant bits into `h.api._compat`. Also the below imports.